File: App/Model/Order.cs
using System.Collections.Generic;
using System.ComponentModel;

namespace App.Model
{
    public class Order_ : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler? PropertyChanged;

        public int Id { get; set; }
        public string InvoiceCode { get; set; }
        public string Customer { get; set; }
        public string SaleDateTime { get; set; }
        private List<Product> orderedProducts { get; set; }
        public decimal TotalAmount { get; set; }
        public decimal TotalDiscount { get; set; }
        public decimal TotalPayment { get; set; }
        public decimal TotalCost { get; set; }
        public string PaymentMethod { get; set; }
        public string Status { get; set; }
        public string PaymentStatus { get; set; }
        public string Notes { get; set; }

        public Order_(int id, string invoiceCode, string customer, string saleDateTime, List<Product> orderedProducts, decimal totalAmount, decimal totalDiscount, decimal totalPayment, decimal totalCost, string paymentMethod, string status, string paymentStatus, string notes)
        {
            Id = id;
            InvoiceCode = invoiceCode;
            Customer = customer;
            SaleDateTime = saleDateTime;
            orderedProducts = orderedProducts;
            TotalAmount = totalAmount;
            TotalDiscount = totalDiscount;
            TotalPayment = totalPayment;
            TotalCost = totalCost;
            PaymentMethod = paymentMethod;
            Status = status;
            PaymentStatus = paymentStatus;
            Notes = notes;
        }

        public Order_() { }
    }
}

File: BE/src/models/Order.js
const mongoose = require("mongoose");

const OrderedProductSchema = new mongoose.Schema({
    ProductCode: { type: String, required: false },
    Name: { type: String, required: true },
    Quantity: { type: Number, required: true },
    Price: { type: Number, required: true }
});

const OrderSchema = new mongoose.Schema(
    {
        InvoiceCode: { type: String, required: true },
        Customer: { type: String, required: true },
        SaleDateTime: { type: Date, required: true, default: Date.now },
        OrderedProducts: { type: [OrderedProductSchema], required: true }, // Danh sách sản phẩm
        TotalAmount: { type: Number, required: true },
        TotalDiscount: { type: Number, required: true },
        TotalPayment: { type: Number, required: true },
        TotalCost: { type: Number, required: true },
        PaymentMethod: { type: String, required: true },
        Status: { type: String, required: true },
        PaymentStatus: { type: String, required: true },
        Notes: { type: String, default: "" }
    },
    { timestamps: true, collection: "order_" } // Đổi collection thành "orders" (số nhiều)
);

const Order = mongoose.model("Order", OrderSchema);
module.exports = Order;
